kerosene/nn/utils/gradients.py

```diff
@@ -0,0 +1,74 @@
+from typing import Union, Callable, Iterable
+
+import torch
+from torch.nn.utils import clip_grad_norm_
+from torch.nn.utils import clip_grad_value_
+
+from enum import Enum
+import abc
+
+
+class GradientClippingStrategyType(Enum):
+    Value = "value"
+    Norm = "norm"
+
+    def __str__(self):
+        return self.value
+
+
+class GradientClippingStrategy(metaclass=abc.ABCMeta):
+
+    @abc.abstractmethod
+    def clip(self):
+        raise NotImplementedError()
+
+
+class GradientNormClipping(GradientClippingStrategy):
+
+    def __init__(self, model_parameters: Union[Iterable[torch.Tensor], torch.Tensor], max_norm: float,
+                 norm_type: Union[int, float]):
+        """"
+        Gradient norm clipping strategy.
+
+        Args:
+            max_norm (float or int): max norm of the gradients
+            norm_type (float or int): type of the used p-norm. Can be ``'inf'`` for infinity norm.
+        """
+        self._model_parameters = model_parameters
```

Comment:
Should put it as a arg to the clip func. A strategy does not need parameters to exists :small_red_triangle: